src/types.ts

```diff
@@ -33,8 +33,10 @@ export interface AlphaOptions<D = any> extends AxiosRequestConfig<D> {
   Lambda?: typeof Lambda;
 }
 
-export type AlphaAdapter = (config: AlphaOptions) => AxiosPromise;
-export type AlphaInterceptor = (config: AlphaOptions) => (Promise<AlphaOptions> | AlphaOptions);
+export type AlphaOptionsForLambda<D = any> = AlphaOptions<D> & InternalAxiosRequestConfig;
+
+export type AlphaAdapter<V = AlphaOptions> = (config: V) => AxiosPromise;
+export type AlphaInterceptor<V = AlphaOptions> = (config: V) => (Promise<V> | V);
```

Comment:
These types don't seem right to me. I think they should just not be generic, and should specify the "internal config" type, not the external one.

```suggestion
export type AlphaAdapter = (config: AlphaOptionsForLambda) => AxiosPromise;
export type AlphaInterceptor = (config: AlphaOptionsForLambda) => (Promise<AlphaOptionsForLambda> | AlphaOptionsForLambda);
```

The main reason why I think they should not be generic: there are no call sites which specify a type _other_ than `AlphaOptionsForLambda` for `V`.

Reply:
The problem is these types are reused for both http request and lambda invocation, so it's kind of `generic`.

the original design is basically following the axios's refactor approach to keep these types as internal config, however, the consequence is it will change existing Alpha API to construct headers as `new AxiosHeaders()` instead of `{}`. see discussion [here](https://github.com/lifeomic/alpha/pull/165#discussion_r1391786093)

Reply:
> The problem is these types are reused for both http request and lambda invocation, so it's kind of generic.

I'm not following this -- in what situation would we specify a type for `V` that is _not_ `AlphaOptionsForLambda`?

---
@shawnzhu jumping from that discussion: there is no need to use `new AxiosHeaders()` at that call site. Replacing it with an empty object `{}` works just fine, with no other changes.

Reply:
You're right, these type changes are for lambda invocation only. see https://github.com/lifeomic/alpha/pull/165/commits/0690f50de5477020461942b0cbaae9e3ea57c47e